staging/src/k8s.io/kubectl/pkg/util/templates/templater.go

```diff
@@ -218,29 +218,84 @@ func (t *templater) usageLine(c *cobra.Command) string {
 	return usage
 }
 
-func flagsUsages(f *flag.FlagSet) string {
-	x := new(bytes.Buffer)
+func flagsUsages(f *flag.FlagSet) (string, error) {
+	flagBuf := new(bytes.Buffer)
+	printer := NewHelpFlagPrinter(flagBuf)
+
+	wrapLimit, err := term.GetWordWrapperLimit()
+	if err != nil {
+		return "", err
+	}
+	minFlagLen := uint(getMinFlagLen(f))
+	maxFlagLen := uint(getMaxFlagLen(f))
+	tabWidth := maxFlagLen - minFlagLen // the maximum width that will be padded by tabWriter
 
 	f.VisitAll(func(flag *flag.Flag) {
 		if flag.Hidden {
 			return
 		}
-		format := "--%s=%s: %s\n"
+		printer.PrintHelpFlags(flag, wrapLimit, tabWidth, minFlagLen, maxFlagLen)
+	})
+	printer.FlushTabWriter()
+	return flagBuf.String(), nil
+}
 
-		if flag.Value.Type() == "string" {
-			format = "--%s='%s': %s\n"
-		}
+func getFlagFormatWithTab(f *flag.Flag) string {
+	var format string
+	format = "--%s=%s:\t%s"
+	if f.Value.Type() == "string" {
+		format = "--%s='%s':\t%s"
+	}
+	if len(f.Shorthand) > 0 {
+		format = "    -%s, " + format
+	} else {
+		format = "     %s   " + format
+	}
+	return format
+}
 
-		if len(flag.Shorthand) > 0 {
-			format = "  -%s, " + format
-		} else {
-			format = "   %s   " + format
-		}
+func getFlagFormatWithoutTab(f *flag.Flag) string {
+	var format string
+	format = "--%s=%s: %s"
+	if f.Value.Type() == "string" {
+		format = "--%s='%s': %s"
+	}
+	if len(f.Shorthand) > 0 {
+		format = "    -%s, " + format
+	} else {
+		format = "     %s   " + format
+	}
+	return format
+}
+
+func getMaxFlagLen(f *flag.FlagSet) int {
+	formatBuf := new(bytes.Buffer)
+	maxFlagLen := 0
 
-		fmt.Fprintf(x, format, flag.Shorthand, flag.Name, flag.DefValue, flag.Usage)
+	f.VisitAll(func(flag *flag.Flag) {
+		formatBuf.Reset()
+		fmt.Fprintf(formatBuf, getFlagFormatWithTab(flag), flag.Shorthand, flag.Name, flag.DefValue, flag.Usage)
+		flagLen := len(formatBuf.String()) - len(flag.Usage)
+		if flagLen > maxFlagLen {
+			maxFlagLen = flagLen
+		}
 	})
+	return maxFlagLen
+}
+
+func getMinFlagLen(f *flag.FlagSet) int {
```

Comment:
Can this and `getMaxFlagLen` become a single `getMinMaxFlagLen` and be handled with a single pass of the flags?